pom.xml

```diff
@@ -1117,6 +1165,54 @@
             </executions>
           </plugin>
           <plugin>
+              <groupId>org.apache.maven.plugins</groupId>
+              <artifactId>maven-dependency-plugin</artifactId>
+              <version>3.0.0</version>
+              <executions>
+                  <!-- TODO as above -->
+                  <execution>
+                      <id>download node for Linux</id>
+                      <phase>initialize</phase>
+                      <goals>
+                          <goal>get</goal>
+                      </goals>
+                      <configuration>
+                          <artifact>com.github.eirslett:node:${node.version}:tar.gz:linux-x64</artifact>
+                      </configuration>
+                  </execution>
+                  <execution>
+                      <id>download node for Darwin</id>
+                      <phase>initialize</phase>
+                      <goals>
+                          <goal>get</goal>
+                      </goals>
+                      <configuration>
+                          <artifact>com.github.eirslett:node:${node.version}:tar.gz:darwin-x64</artifact>
+                      </configuration>
+                  </execution>
+                  <execution>
+                      <id>download node for Windows</id>
+                      <phase>initialize</phase>
+                      <goals>
+                          <goal>get</goal>
+                      </goals>
+                      <configuration>
+                          <artifact>com.github.eirslett:node:${node.version}:exe:windows-x64</artifact>
+                      </configuration>
+                  </execution>
+                  <execution>
+                      <id>download yarn</id>
+                      <phase>initialize</phase>
+                      <goals>
+                          <goal>get</goal>
+                      </goals>
+                      <configuration>
+                          <artifact>com.github.eirslett:yarn:${yarn.version}:tar.gz</artifact>
```

Comment:
These changes seem reasonable. I agree that `node/npm/yarnDownloadRoot` would be much better to use. Is it worth digger deeper into the possibilities in Artifactory config? That would be much less configuration to put in this POM file... although I guess ultimately the config burden needs to go either here or in there, and maybe it's better not to assume that some exotic Artifactory config is in place?